featuretools/synthesis/deep_feature_synthesis.py

```diff
@@ -299,7 +300,7 @@ def _run_dfs(self, entity, entity_path, all_features, max_depth):
         """
 
         backward_entities = self.es.get_backward_entities(entity.id)
-        backward_entities = [b_id for b_id in backward_entities
+        backward_entities = [b_id for b_id, _ in backward_entities
```

Comment:
If an entity has two backward paths to a child (e.g teams -> games.home_team, teams -> games.away_team), would the child show up twice in this list? 

if it does, we'd end up making the same call to `_run_dfs` twice. I imagine this doesn't cause problems since it just wouldn't add same feature twice, but probably not ideal.

Reply:
You are right. And there is even a check for this, but it never gets run: https://github.com/Featuretools/featuretools/blob/1af738fdc77d65e177b66412cfe4a577c509d0df/featuretools/synthesis/deep_feature_synthesis.py#L409-L412

This is tangential to this PR, but should we remove the `return` to enforce this? There are several tests which would hit that exception. This is because the `variable_type` of `LAST(log.id)` is `Index`, so DFS on `customers` attempts to create two `COUNT(sessions)` features, one where the base feature is `LAST(log.id)` and the other where the base feature is `sessions.id`. Should the `variable_type` of `LAST(log.id)` be `Numeric` instead?

Reply:
Yeah, we should handle this outside of this PR. Some notes below

In fact, `LAST(log.id)` shouldn't be `Index` -- that is a bug. The `return_type` of `Last` is `None`, it should trigger the logic here

https://github.com/Featuretools/featuretools/blob/1af738fdc77d65e177b66412cfe4a577c509d0df/featuretools/feature_base/feature_base.py#L173

We already handle the timeindex case, but we should also handle the index case and just return that it is `Categorical`. 

Can you put up an issue to handle later, or PR and test case?

We can continue discuss on the new Issue/PR, but if we fix that, does any other test case raise the error?

Reply:
I created #602 to fix that.

There is one other case where this is hit: when there is a `seed_feature` that is also generated by `dfs`.